### DISCUSSION TITLE
Help! How to use sx values to dynamically change styling of a component in Storybook UI

### DISCUSSION THREAD

**Discussion by daryllmamaril** [storybookjs/storybook#26851] [Help] [answered]
### Summary

Hey everyone!
I've just started using Storybook so I hope my question will make sense.
I have a Box Component. And for the Box.stories.tsx I have this file:

```const meta: Meta<IBoxProps> = {
  title: "Components/Box",
  component: Box,
  argTypes: {
    children: {
      name: "Text",
      control: "text",
    },
    sx: {
      control: "object"
    }
  },
    },
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Box>;

export default meta;

type Story = StoryObj<typeof meta>;

export const WithMaxWidth: Story = {
  args: {
    children:
      "This Box renders as an HTML main element. The width of the box has been set with a maximum of 10rem.",
  },
  render: (args) => {
    return (
      <Box
        {...args}
        sx={{
          backgroundColor: "#F3F5F9",
          border: "1px solid #4E545C",
          maxWidth: "10rem",
        }}
      />
    );
  },
};
```
How can I use the sx on my WithMaxWidth story --> (sx={{backgroundColor: "#F3F5F9", border: "1px solid #4E545C", maxWidth: "10rem"}} as a control on storybook(default)? For example I want to have a control with the name sx with an argType of "text" or "object" or whatever is most suitable. And when someone edits the sx styling in that textbox, the component rerenders according to the sx values that was entered. Is that possible? I hope the question makes sense.
![question](https://github.com/storybookjs/storybook/assets/74665101/4320663e-09ef-46e6-88a8-8c12b2b4e419)

**Comment by Marklb** [comment, marked as answer]
I think I understand your question, but I am not positive. Are you saying that you already know how to add the control, which is shown in your snippet and screenshot, but you are not sure how to use it?

Controls are going to be created from the `argTypes`, with the `args` as the values. When rendering, the values are passed to the story as `args` again. The `render` function receives the `args` as it's first parameter, which you spread as the component inputs with `<Box {...args}`. What you currently pass to the `sx` input is what I assume you want as the initial value.

Assuming I understood your question, the following is how your story should be changed to. *The following snippet is a minor change from the original, but the next snippet is more simplified.*

```tsx
export const WithMaxWidth: Story = {
  args: {
    children:
      "This Box renders as an HTML main element. The width of the box has been set with a maximum of 10rem.",
    sx: {
      backgroundColor: "#F3F5F9",
      border: "1px solid #4E545C",
  	  maxWidth: "10rem",
    },
  },
  render: (args) => {
    return (
      <Box
        {...args}
        sx={args.sx}
      />
    );
  },
};
```

You should be able to remove `sx={args.sx}` or just remove the `render` method, since `sx` is in args.

```tsx
export const WithMaxWidth: Story = {
  args: {
    children:
      "This Box renders as an HTML main element. The width of the box has been set with a maximum of 10rem.",
    sx: {
      backgroundColor: "#F3F5F9",
      border: "1px solid #4E545C",
  	  maxWidth: "10rem",
    },
  },
};
```

**Reply by daryllmamaril** [reply]
@Marklb Thanks so much! This works fine for now. I think in the future I need to make the sx control as a string so that it can be passed to sitecore. But I guess that's a problem for another day. Thanks again for your help. Really appreciate it! :)
